rpcwebsocket.go

```diff
@@ -2035,802 +2079,146 @@ func deserializeOutpoints(serializedOuts []dcrjson.OutPoint) ([]*wire.OutPoint,
-				if err == ErrClientQuit {
-					return
-				}
-				spentNotified = true
 			}
 		}
+	}
 
-		for txOutIdx, txout := range tx.MsgTx().TxOut {
-			_, addrs, _, _ := txscript.ExtractPkScriptAddrs(txout.Version,
-				txout.PkScript, wsc.server.server.chainParams)
-
-			for _, addr := range addrs {
-				switch a := addr.(type) {
-				case *dcrutil.AddressPubKeyHash:
-					if _, ok := lookups.pubKeyHashes[*a.Hash160()]; !ok {
-						continue
-					}
-
-				case *dcrutil.AddressScriptHash:
-					if _, ok := lookups.scriptHashes[*a.Hash160()]; !ok {
-						continue
-					}
-
-				case *dcrutil.AddressSecpPubKey:
-					found := false
-					switch sa := a.ScriptAddress(); len(sa) {
-					case 33: // Compressed
-						var key [33]byte
-						copy(key[:], sa)
-						if _, ok := lookups.compressedPubKeys[key]; ok {
-							found = true
-						}
-
-					case 65: // Uncompressed
-						var key [65]byte
-						copy(key[:], sa)
-						if _, ok := lookups.uncompressedPubKeys[key]; ok {
-							found = true
-						}
-
-					default:
-						rpcsLog.Warnf("Skipping rescanned pubkey of unknown "+
-							"serialized length %d", len(sa))
-						continue
-					}
-
-					// If the transaction output pays to the pubkey of
-					// a rescanned P2PKH address, include it as well.
-					if !found {
-						pkh := a.AddressPubKeyHash()
-						if _,
-							ok := lookups.pubKeyHashes[*pkh.Hash160()]; !ok {
-							continue
-						}
-					}
-
-				default:
-					// A new address type must have been added.  Encode as a
-					// payment address string and check the fallback map.
-					addrStr := addr.EncodeAddress()
-					_, ok := lookups.fallbacks[addrStr]
-					if !ok {
-						continue
-					}
-				}
-
-				var outpoint wire.OutPoint
-				if tx.Tree() == dcrutil.TxTreeRegular {
-					outpoint = wire.OutPoint{
-						Hash:  *tx.Sha(),
-						Index: uint32(txOutIdx),
-						Tree:  dcrutil.TxTreeRegular, // decred
-					}
-				} else if tx.Tree() == dcrutil.TxTreeStake {
-					outpoint = wire.OutPoint{
-						Hash:  *tx.Sha(),
-						Index: uint32(txOutIdx),
-						Tree:  dcrutil.TxTreeStake, // decred
-					}
-				}
-				lookups.unspent[outpoint] = struct{}{}
-
-				if recvNotified {
-					continue
-				}
-
-				if txHex == "" {
-					txHex = txHexString(tx)
-				}
-
-				var ntfn *dcrjson.RecvTxNtfn
-				ntfn = dcrjson.NewRecvTxNtfn(txHex, blockDetails(nil, tx.Tree(),
-					tx.Index()))
-
-				marshalledJSON, err := dcrjson.MarshalCmd(nil, ntfn)
-				if err != nil {
-					rpcsLog.Errorf("Failed to marshal recvtx "+
-						"notification: %v", err)
-					return
-				}
-
-				err = wsc.QueueNotification(marshalledJSON)
-				// Stop the rescan early if the websocket client
-				// disconnected.
-				if err == ErrClientQuit {
-					return
-				}
-				recvNotified = true
-			}
-		}
+	msgBlock := block.MsgBlock()
+	filter.mu.Lock()
+	for _, tx := range msgBlock.STransactions {
+		checkTransaction(tx, dcrutil.TxTreeStake)
 	}
+	for _, tx := range msgBlock.Transactions {
```

Comment:
I disagree with that design decision.  Wallet will save them regardless and handle the case when they are invalidated by stake voters in the next block.